Standard/src/Preparation/Arbitrary.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections;
+using System.Runtime.InteropServices;
+using Microsoft.Quantum.Simulation;
+using Microsoft.Quantum.Simulation.Core;
+using Microsoft.Quantum.Simulation.Simulators;
+
+
+namespace Microsoft.Quantum.Preparation
+{
+    public partial class PrepareArbitraryState
+    {
+        /// <summary>
+        ///  Provides a native emulation of the PrepareArbitraryState operation when
+        ///  the operation is executed using the full-state QuantumSimulator.
+        /// </summary>
+        public class Native : PrepareArbitraryState
+        {
+            [DllImport(QuantumSimulator.QSIM_DLL_NAME, ExactSpelling = true,
+                CallingConvention = CallingConvention.Cdecl, EntryPoint = "InjectState")]
+            private static extern double InjectState(uint sid, uint n, uint[] q, double[] re, double[] im);
+
+            private QuantumSimulator Simulator { get; }
```

Comment:
Since this can be null, I'd suggest turning on `#nullable enable` and modifying the type of this field to denote nullability:

```suggestion
            private QuantumSimulator? Simulator { get; }
```